### ISSUE TITLE
Focus indicator is not shown when using Checkbox inside an element with overflow set to hidden

### ISSUE THREAD

**Issue by Lijing29** [OfficeDev/office-ui-fabric-react#11854]
### Environment Information

- **Package version(s)**: 7
- **Browser and OS versions**: 
Chrome: 79.0.3945.130
Edge: 44.18362.449.0

### Describe the issue:

When using Checkbox component as a child of an element, such div, with style overflow = hidden, the focus indicator would not be shown when using tab navigating it.

### Please provide a reproduction of the issue in a codepen: 
https://codepen.io/lijinglin29/pen/ExaqExJ

#### Actual behavior:
When using Checkbox component as a child of an element, such div, with style overflow = hidden, the focus indicator would not be shown when using tab navigating it.

#### Expected behavior:

The focus indicator should be shown when using tab navigating it.

**Comment by xugao** [comment]
@Lijing29 - in this case, you will need to add some padding on the parent element or add some margin on the checkbox root element. here is a demo: https://codepen.io/xugao/pen/ExaqJvZ

**Comment by Lijing29** [comment]
Thanks for the suggestion. The method you proposed works. But it is not a good work around in some of the situation, such as a list of checkbox as adding additional padding or margin may make the UX strange.

Currently, the outline-offset is set to 2px of a label inside the checkbox.
![image](https://user-images.githubusercontent.com/50400285/73708794-5cdec500-473a-11ea-9254-ca19bd7059f1.png)
I guess if we do not set the outline-offset=2px of checkbox-label, checkbox may work well inside an element with overflow set to hidden.

**Comment by xugao** [comment]
@Lijing29 - there is not much we can do since you are setting `overflow: hidden`. we cannot just remove the `offset` here. This is a case you will have to overwrite the styles based on your case. 
Not perfect, but this is what you can do:
```
  import { IsFocusVisibleClassName } from '@uifabric/utilities';

  const checkboxStyles: ICheckboxStyles = {
    input: {
      selectors: {
        [`.${IsFocusVisibleClassName} &:focus + label::before`]: {
          outlineOffset: '0px'
        }
      }
    }
  };

  <Checkbox styles={checkboxStyles} label="Unchecked checkbox (uncontrolled)" />
```
